src/Word/CamelCaseToSeparator.php

```diff
@@ -15,23 +16,24 @@
  *     ...
  * }
  * @template TOptions of Options
- * @extends AbstractSeparator<TOptions>
+ * @implements FilterInterface<string|array<array-key, string|mixed>>
  */
-class CamelCaseToSeparator extends AbstractSeparator
+class CamelCaseToSeparator implements FilterInterface
```

Comment:
I'd like this to be final and for the 2 descendants to compose this class instead